src/notification.c

```diff
@@ -228,6 +228,7 @@ handle_add_notification (XdpImplNotification *object,
                          GVariant *arg_notification)
 {
   if (gtk_notifications == NULL ||
+      g_str_equal (arg_app_id, "") ||
```

Comment:
Should it just be `!g_application_id_is_valid (arg_app_id)`

Reply:
That seems clearer, thanks!